lib/osf-components/addon/components/registries/update-dropdown/component.ts

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable no-console */
+import { inject as service } from '@ember/service';
+import { waitFor } from '@ember/test-waiters';
+import Component from '@glimmer/component';
+import { task } from 'ember-concurrency';
+import Intl from 'ember-intl/services/intl';
+import { QueryHasManyResult } from 'ember-osf-web/models/osf-model';
+
+import RegistrationModel from 'ember-osf-web/models/registration';
+import RevisionModel, { RevisionReviewStates } from 'ember-osf-web/models/revision';
+import CurrentUserService from 'ember-osf-web/services/current-user';
+import Toast from 'ember-toastr/services/toast';
+import captureException, { getApiErrorMessage } from 'ember-osf-web/utils/capture-exception';
+import Store from '@ember-data/store';
+import RouterService from '@ember/routing/router-service';
+import { taskFor } from 'ember-concurrency-ts';
+import { computed } from '@ember/object';
+import registration from 'ember-osf-web/mirage/factories/registration';
+
+interface Args {
+    registration: RegistrationModel;
+}
+
+type RevisionJustification = 'Adding Results' | 'Typo - Self' | 'Typo - Other' | 'Copy Edit';
+
+export default class UpdateDropdown extends Component<Args> {
+    @service currentUser!: CurrentUserService;
+    @service intl!: Intl;
+    @service store!: Store;
+    @service toast!: Toast;
+    @service router!: RouterService;
+
+    revisions?: QueryHasManyResult<RevisionModel>;
+    isPendingCurrentUserApproval?: boolean;
+    revisionJustification?: RevisionJustification;
+
+    constructor(owner: unknown, args: Args) {
+        super(owner, args);
+        taskFor(this.getRevisionList).perform();
+    }
+
+    @computed('args.registration.{userHasAdminPermission,revisionState}')
+    get shouldDisplayApproveDenyButtons() {
+        return this.args.registration.userHasAdminPermission
+        && this.args.registration.revisionState
+        && ![
+            RevisionReviewStates.RevisionInProgress,
+            RevisionReviewStates.Approved,
+        ].includes(this.args.registration.revisionState);
+    }
+
+    @computed('args.registration.revisionState')
+    get updateNotificationIcon() {
+        switch (this.args.registration.revisionState) {
+        case RevisionReviewStates.Approved:
+            return 'lock';
+        case RevisionReviewStates.RevisionInProgress:
+            return 'eye';
+        case RevisionReviewStates.RevisionPendingAdminApproval:
+        case RevisionReviewStates.RevisionPendingModeration:
+            return 'clock';
+        default:
+            return '';
+        }
+    }
+
+    @task
+    @waitFor
+    async getRevisionList() {
+
+        if (!registration){
+            const notRegistrationError = this.intl.t('registries.update_dropdown.not_a_registration_error');
+            return this.toast.error(notRegistrationError);
+        }
+        if (this.args.registration.revisions === null || this.args.registration.revisions === undefined) {
```

Comment:
Got it, yes. I may ask you or Futa again why this would not work. I was mentored that 'if (!this.args.registration)' would be enough to catch / account for null or undefined values but would like to confirm the logic.